op-node/rollup/driver/driver.go

```diff
@@ -167,8 +179,12 @@ func NewDriver(
 	plasma PlasmaIface,
 ) *Driver {
 	driverCtx, driverCancel := context.WithCancel(context.Background())
-	rootDeriver := &rollup.SynchronousDerivers{}
-	synchronousEvents := rollup.NewSynchronousEvents(log, driverCtx, rootDeriver)
+	rootDeriver := &event.DeriverMux{}
+	var synchronousEvents event.EmitterDrainer
+	synchronousEvents = event.NewQueue(log, driverCtx, rootDeriver, metrics)
+	synchronousEvents = event.NewLimiterDrainer(context.Background(), synchronousEvents, eventsLimit, eventsBurst, func() {
+		log.Warn("Driver is hitting events rate limit.")
+	})
```

Comment:
I'd suggest we add a metric for this - just a counter of the number of times the limit was hit is probably enough. We probably should setup alerting for it since it's not something we generally expect to happen. Theoretically we could do that by looking at the rate of increase of the event count metric but would be good to have an at-a-glance number too.

Although, if we're doing something like syncing from genesis would we hit this limit or is the delay for geth to import the block long enough to keep us below it?

Reply:
Added metrics. And bumped the rate-limit to 10k events per second. If we have 2ms blocks (already unreal, avg block is more like 20-100ms), then we see 500 blocks / second. But we do process multiple events per block, so bumping it to 10k makes sense.